src/Compositor.cpp

```diff
@@ -3039,31 +3040,45 @@ void CCompositor::ensurePersistentWorkspacesPresent(const std::vector<SWorkspace
             continue;
         }
 
-        WORKSPACEID id     = rule.workspaceId;
-        std::string wsname = rule.workspaceName;
-        if (id == WORKSPACE_INVALID) {
-            const auto R = getWorkspaceIDNameFromString(rule.workspaceString);
-            id           = R.id;
-            wsname       = R.name;
+        PHLWORKSPACE PWORKSPACE = nullptr;
+        if (pWorkspace) {
```

Comment:
this check can be moved above PMONITOR finding a monitor for better optimization